### PR TITLE
RSFB-3327/Concat2 Function Unparsing related changes

### PR DESCRIPTION
**Pull request by ashutosh-kumar2229**
Resolving extra t in case of concat in postgres procedure which was coming due to extra boolean Operand added to differentiate between CONCAT Operator(||) and CONCAT Function.

### CODE REVIEW

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java [hunk 1]
```diff
@@ -3386,6 +3386,20 @@ private void checkHavingAliasSameAsColumn(boolean upperAlias) {
     assertThat(toSql(root, DatabaseProduct.BIG_QUERY.getDialect()), isLinux(expectedBiqQuery));
   }
 
+  @Test public void testConcat2FunctionWithBooleanArgumentRelToSql() {
```
Comment (naveenantonie) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java:3389]: need not mention 'RelToSql' explicitly

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java [hunk 2]
```diff
@@ -3386,6 +3386,20 @@ private void checkHavingAliasSameAsColumn(boolean upperAlias) {
...
+    final RelBuilder builder = relBuilder();
+    final RexNode concatRexNode =
+        builder.call(SqlLibraryOperators.CONCAT2, builder.literal("foo"),
+            builder.literal("bar"), builder.literal("true"));
+    final RelNode root = builder
+        .scan("EMP")
+        .project(builder.alias(concatRexNode, "CR"))
+        .build();
+    final String expectedSql = "SELECT CONCAT('foo', 'bar') AS \"CR\"\n"
```
Comment (naveenantonie) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java:3398]: we could add an assertion for sqlNode unparsing as well

```
final String expectedSql = "SELECT CONCAT('foo', 'bar', 'true') AS \"CR\"\n"
        + "FROM \"scott\".\"EMP\"";
    final String expectedPostgresSql = "SELECT CONCAT('foo', 'bar') AS \"CR\"\n"
        + "FROM \"scott\".\"EMP\"";
    assertThat(toSql(root, DatabaseProduct.CALCITE.getDialect()), isLinux(expectedSql));
    assertThat(toSql(root, DatabaseProduct.POSTGRESQL.getDialect()), isLinux(expectedPostgresSql));
```

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java (hunk 1 above)
Comment (ashutosh-kumar2229) [reply core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java:3389]: removed

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java (hunk 2 above)
Comment (ashutosh-kumar2229) [reply core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java]: added

**Comment by naveenantonie** [comment]
Looks good to me!

**Review by anuragsaxena2k7 — APPROVED** [review]
approve

**Pull request merged**